mlperf_logging/rcp_checker/rcp_checker.py

```diff
@@ -148,6 +148,57 @@ def _process_raw_rcp_data(self, raw_rcp_data):
         return processed_rcps
 
 
+    def _prune_rcps(self):
+        '''
+        Prune RCPs. We compare convergence of each RCP point with interpolation usgin surrounding points
+        and move RCP points that have min (fastest) convergence to min_rcp_data.
+        min_rcp_data is by default used for RCP tests.
+        '''
+        self.min_rcp_data = {}
+        for benchmark in submission_runs['training'].keys():
+            min_epochs = []
+            for record, record_contents in self.rcp_data.items():
+                if record_contents['Benchmark'] == benchmark:
+                    min_epochs.append(record_contents)
+
+            # Step 1
+            # Find point with fastest convergence and prune all point with smaller batch size
+            # In that way the min batch size point will have the fastest convergenece
+            fastest_conv = min_epochs[0]['Min Epochs']
+            fastest_conv_idx = 0
+            for i, min_epoch in enumerate(min_epochs):
+                if min_epoch['Min Epochs'] < fastest_conv:
+                    fastest_conv_idx = i
+                    fastest_conv = min_epochs[i]['Min Epochs']
+            # print('FASTEST CONV:', benchmark, fastest_conv_idx)
+            for i in range(fastest_conv_idx):
+                del min_epochs[0]
+
+            # Step 2
+            # Run this algorithm for the rest of the points:
+            # for i = 1..N-2
+            #    if RCP[i+1] has slower convergence than interpolation(RCP[i], RCP[i+2]):
+            #      remove it
+            #      decrement i,N
+            list_len = len(min_epochs)
+            i = 1
+            while i < list_len - 1:
+                rcp_min = min_epochs[i-1]
+                rcp_max = min_epochs[i+1]
+                bs = min_epochs[i]['BS']
+                name, rcp = self._create_interp_rcp(benchmark,bs,rcp_min,rcp_max,show_rec=False)
+                if min_epochs[i]['Min Epochs'] > rcp['Min Epochs']:
+                    del min_epochs[i]
+                    i = i-1
+                    list_len = list_len - 1
+                i = i+1
+
+            # Copy pruned epochs to min_rcp_data
+            for min_epoch in min_epochs:
```

Comment:
Move this part outside to the `for benchmark in submission_runs['training'].keys():` loop and
```python
records_to_keep = {
  benchmark: set(rc['BS'] for rc in min_epochs) 
  for benchmark, min_epochs in min_epochs_per_benchmark.items()
}
self.min_rcp_data = dict(filter(lambda r, rc: rc['BS'] in records_to_keep[rc['Benchmark']], self.rcp_data))
```

Reply:
Since I did not change the previous one, I am not changing this either.